xbmc/addons/AddonManager.cpp

```diff
@@ -671,7 +677,11 @@ bool CAddonMgr::FindAddons()
       for (int i = 0; i < n; ++i)
         installed.insert(cp_addons[i]->identifier);
       m_cpluff->release_info(m_cp_context, cp_addons);
-      m_database.SyncInstalled(installed, m_systemAddons);
+
+      std::set<std::string> addonsToEnable(m_systemAddons);
+      addonsToEnable.insert(m_optionalAddons.begin(), m_optionalAddons.end());
+
+      m_database.SyncInstalled(installed, addonsToEnable);
```

Comment:
This will set them to enabled in db. I.e your disabled optional addons will be re-enabled on every sync.